run.c

```diff
@@ -195,14 +195,17 @@ void matmul(float* xout, float* x, float* w, int n, int d) {
     // W (d,n) @ x (n,) -> xout (d,)
     // by far the most amount of time is spent inside this little function
     int i;
-    #pragma omp parallel for private(i)
+    #pragma omp target data map(to:x[0:n]) map(from:xout[0:d])
+    #pragma omp target teams distribute
     for (i = 0; i < d; i++) {
         float val = 0.0f;
+        #pragma parallel for reduction(+:val) 
```

Comment:
Noticing a bug here. You are missing an `omp` in the directive. It should be `#pragma omp parallel for` not just `#pragma parallel for`. Also just a side note, using `#pragma omp simd` for the 2nd loop yielded much better results for me.